nifi-commons/nifi-expression-language/src/main/java/org/apache/nifi/attribute/expression/language/evaluation/functions/GeohashLongEncodeEvaluator.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.nifi.attribute.expression.language.evaluation.functions;
+
+import org.apache.nifi.attribute.expression.language.EvaluationContext;
+import org.apache.nifi.attribute.expression.language.evaluation.Evaluator;
+import org.apache.nifi.attribute.expression.language.evaluation.QueryResult;
+import org.apache.nifi.attribute.expression.language.evaluation.WholeNumberEvaluator;
+import org.apache.nifi.attribute.expression.language.evaluation.WholeNumberQueryResult;
+import org.apache.nifi.attribute.expression.language.exception.AttributeExpressionLanguageException;
+
+import ch.hsr.geohash.GeoHash;
+
+public class GeohashLongEncodeEvaluator extends WholeNumberEvaluator {
+
+    private final Evaluator<Number> latitude;
+    private final Evaluator<Number> longitude;
+    private final Evaluator<Long> level;
+
+    public GeohashLongEncodeEvaluator(final Evaluator<Number> latitude, final Evaluator<Number> longitude, final Evaluator<Long> level) {
+        this.latitude = latitude;
+        this.longitude = longitude;
+        this.level = level;
+    }
+    @Override
+    public QueryResult<Long> evaluate(final EvaluationContext evaluationContext) {
+        final Number latitudeValue = latitude.evaluate(evaluationContext).getValue();
+        if (latitudeValue == null) {
```

Comment:
If we do not receive a valid value for latitude/longitude it seems as if no error/bulletin is appearing in NiFi.  I think we should raise an exception, otherwise users would not be alerted in cases where data might be bad or corrupted.

Reply:
I’ve changed it to throw exceptions when lat/lon provided are invalid, but null lat/lon values are still allowed and the functions will return empty geohashes in this case. We throw exceptions if empty or invalid level values are provided.

Reply:
After testing this behavior I still recommend that we throw an exception if latitude and longitude are also null.  In the tested scenario  with the below data:
```
{"latitude": 27.2046,"longitude": "cat"}
```
I would expect the processor to report an error however it does not. In current form the evaluator looks to determine whether the provided string is actually a string form of a number, however if it's not a number it returns null. I think this should be an invalid case and users should be alerted.  

I also think that in the case where one or the other or even both values were truly provided as null it should also treat it as an exception. Users leveraging this function could provide a check in a previous flow step if need to divert data that may fail. I would assume that if someone calls this function their intent is for valid lat/lons to be available.